samples/net/mqtt_azure/src/main.c

```diff
@@ -0,0 +1,482 @@
+	inet_pton(AF_INET, SERVER_ADDR, &broker4->sin_addr);
+
+#if defined(CONFIG_SOCKS)
+	struct sockaddr_in *proxy4 = (struct sockaddr_in *)&socks5_proxy;
+
+	proxy4->sin_family = AF_INET;
+	proxy4->sin_port = htons(SOCKS5_PROXY_PORT);
+	inet_pton(AF_INET, SOCKS5_PROXY_ADDR, &proxy4->sin_addr);
+#endif
+}
+
+static void client_init(struct mqtt_client *client)
+{
+	const char azure_iot_username[] = CONFIG_SAMPLE_CLOUD_AZURE_USERNAME;
+	const char azure_iot_password[] = CONFIG_SAMPLE_CLOUD_AZURE_PASSWORD;
+
+	static struct mqtt_utf8 password;
+	static struct mqtt_utf8 username;
+	struct mqtt_sec_config *tls_config;
+
+	mqtt_client_init(client);
+
+	broker_init();
+
+	/* MQTT client configuration */
+	client->broker = &broker;
+	client->evt_cb = mqtt_event_handler;
+
+	client->client_id.utf8 = (u8_t *)MQTT_CLIENTID;
+	client->client_id.size = strlen(MQTT_CLIENTID);
+
+	password.utf8 = (u8_t *)azure_iot_password;
+	password.size = strlen(azure_iot_password);
+
+	client->password = &password;
+
+	username.utf8 = (u8_t *)azure_iot_username;
+	username.size = strlen(azure_iot_username);
+
+	client->user_name = &username;
+
+	client->protocol_version = MQTT_VERSION_3_1_1;
+
+	/* MQTT buffers configuration */
+	client->rx_buf = rx_buffer;
+	client->rx_buf_size = sizeof(rx_buffer);
+	client->tx_buf = tx_buffer;
+	client->tx_buf_size = sizeof(tx_buffer);
+
+	/* MQTT transport configuration */
+	client->transport.type = MQTT_TRANSPORT_SECURE;
+
+	tls_config = &client->transport.tls.config;
+
+	tls_config->peer_verify = 2;
+	tls_config->cipher_list = NULL;
+	tls_config->sec_tag_list = m_sec_tags;
+	tls_config->sec_tag_count = ARRAY_SIZE(m_sec_tags);
+#if defined(MBEDTLS_X509_CRT_PARSE_C)
+	tls_config->hostname = TLS_SNI_HOSTNAME;
+#else
+	tls_config->hostname = NULL;
+#endif
+
+#if defined(CONFIG_SOCKS)
+	mqtt_client_set_proxy(client, &socks5_proxy,
+			      socks5_proxy.sa_family == AF_INET ?
+			      sizeof(struct sockaddr_in) :
+			      sizeof(struct sockaddr_in6));
+#endif
+}
+
+static void mqtt_event_handler(struct mqtt_client *const client,
+			       const struct mqtt_evt *evt)
+{
+	switch (evt->type) {
+	case MQTT_EVT_CONNACK:
+		if (evt->result) {
+			LOG_ERR("MQTT connect failed %d", evt->result);
+			break;
+		}
+
+		mqtt_connected = true;
+		LOG_DBG("MQTT client connected!");
+		break;
+
+	case MQTT_EVT_DISCONNECT:
+		LOG_DBG("MQTT client disconnected %d", evt->result);
+
+		mqtt_connected = false;
+		clear_fds();
+		break;
+
+	case MQTT_EVT_PUBACK:
+		if (evt->result) {
+			LOG_ERR("MQTT PUBACK error %d", evt->result);
+			break;
+		}
+
+		LOG_DBG("PUBACK packet id: %u\n", evt->param.puback.message_id);
+		break;
+
+	default:
+		LOG_DBG("Unhandled MQTT event %d", evt->type);
+		break;
+	}
+}
+
+static int publish(struct mqtt_client *client, enum mqtt_qos qos)
+{
+	char payload[9] = "{id=123}";
+	struct mqtt_publish_param param;
+	char topic[40];
+	u8_t len;
+
+	len = snprintk(topic, sizeof(topic), "devices/%s/messages/events/",
```

Comment:
This leaves up to 15 characters for device_id, probably that's enough, but we could actually avoid this limitation, by initializing `topic` variable statically:
```
	char topic[] = "devices/" CONFIG_SAMPLE_CLOUD_AZURE_CLIENT_ID "/messages/events/";
	u8_t len = strlen(topic);
```